### PR TITLE
feat(article): add scroll timeline control animation blog

### PR DESCRIPTION
**Pull request by SteveJonk**

### CODE REVIEW

File: data/blog/controlling-video-scroll.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+---
+title: 'Controlling your video animations with _scroll-timeline_ and _animation-timeline_'
+date: '2024-09-02'
+tags: ['frontend']
+images: ['/articles/animations-scroll-timeline/airpods.jpg']
+summary: "Step into a (very near) future where you can control animations with pure CSS and without the need for any JavaScript? In this article, I'll show you how to control your animations with the new CSS properties _scroll-timeline_ and _animation-timeline_."
```
Comment (DaveBitter) [inline data/blog/controlling-video-scroll.md]: The underscores don't work in the summary :) See preview url

File: data/blog/controlling-video-scroll.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+The gist of it is actually:
+
+1. Split a video into separate frames and store them as jpg’s. There are a bunch of tools for this, but I used [https://ezgif.com/video-to-jpg](https://ezgif.com/video-to-jpg).
+2. Create a CSS animation to change the background image of an element to the next frame jpg file at every step
+3. The new and fun part: control that CSS animation with a scroll-timeline.
+
+I’ll explain steps 2 and 3 of this short summary with code examples. Don’t worry, the full repo of this demo is shared at the bottom of the article.
+
+**Create the CSS animation**
```
Comment (DaveBitter) [inline data/blog/controlling-video-scroll.md]: Should this (and the others) be 
```suggestion
#### Create the CSS animation**
```

File: data/blog/controlling-video-scroll.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  75% {
+    background-image: url('frame-3.jpg');
+  }
+  100% {
+    background-image: url('frame-4.jpg');
+  }
+}
+```
+
+The magic all happens in the two CSS properties that this whole article is about:
```
Comment (DaveBitter) [inline data/blog/controlling-video-scroll.md:154]: You can highlight these specific two in the codeblock by adding (for example) ```css {10,23} I believe

File: data/blog/controlling-video-scroll.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+- To keep things CSS, I didn't use the SCSS function for the @keyframes
+
+### A possible improvement
+
+A small issue with this approach was that at initial pageload, the images still needed to load when the user scrolled, causing flickering in the video. The solution I implemented probably isn’t the best for production environments. To prevent this from happening, I added a script that preloads all the images. This might need some improvement, but that was not in scope for this demo.
+
+## Final Remarks
+
+I really really love this css feature. It performs very well, I dare to say even a lot better than most JavaScript implementations I have seen. I can’t wait for it to be widely supported!
+
```
Comment (DaveBitter) [inline data/blog/controlling-video-scroll.md:174]: Maybe repeat the gif here with the end result

File: data/blog/controlling-video-scroll.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+title: 'Controlling your video animations with _scroll-timeline_ and _animation-timeline_'
+date: '2024-09-02'
+tags: ['frontend']
+images: ['/articles/animations-scroll-timeline/airpods.jpg']
+summary: "Step into a (very near) future where you can control animations with pure CSS and without the need for any JavaScript? In this article, I'll show you how to control your animations with the new CSS properties _scroll-timeline_ and _animation-timeline_."
+authors: ['steve-jonk']
+theme: 'green'
+---
+
+Our browsers are always evolving and cool new features are added with almost every update. I played around with one of these new features and I love it! I’m talking about the new CSS properties called [scroll-timeline](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-timeline) and [animation-timeline](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-timeline).
```
Comment (DaveBitter) [inline data/blog/controlling-video-scroll.md:11]: Maybe add the part that you worked on it during a hackathon/our google day which is nice exposure for potential candidates

File: data/blog/controlling-video-scroll.md (hunk 1 above)
Comment (SteveJonk) [reply data/blog/controlling-video-scroll.md]: Woops

File: data/blog/controlling-video-scroll.md (hunk 4 above)
Comment (SteveJonk) [reply data/blog/controlling-video-scroll.md:174]: Good one

File: data/blog/controlling-video-scroll.md (hunk 3 above)
Comment (SteveJonk) [reply data/blog/controlling-video-scroll.md:154]: Nice, never knew!

File: data/blog/controlling-video-scroll.md (hunk 5 above)
Comment (SteveJonk) [reply data/blog/controlling-video-scroll.md:11]: GOod one, done!

**Pull request merged**